### PR TITLE
Adds libdl to LD_PRELOAD

### PR DESCRIPTION
**Pull request by nkremerh**
Adds libdl to LD_PRELOAD for applications that do not have this (needed to make dlsym work)

Adds extra checking to beginning of envtrace-helper

### CODE REVIEW

File: envtrace [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
 ENVTRACE_ROOT="$(readlink -f $(dirname "$0"))"
 
-export LD_PRELOAD=${ENVTRACE_ROOT}/envtrace-helper.so
+export LD_PRELOAD=${ENVTRACE_ROOT}/envtrace-helper.so:/lib64/libdl.so.2:${LD_PRELOAD}
```
Comment (dthain) [inline envtrace]: `/lib64/libdl.so.2` seems like a really specific path and version that may break in future upgrades.
Does plain `libdl.so` work?

File: envtrace-helper.c [hunk 2]
```diff
@@ -1,4 +1,11 @@
+#if defined(__linux__) && !defined(_GNU_SOURCE)
 #define _GNU_SOURCE
+#endif
+
+#if !defined(RTLD_NEXT)
+#define RTLD_NEXT 0
```
Comment (dthain) [inline envtrace-helper.c]: This looks fragile to me.

`man dlsym()` says:

```
The symbols RTLD_DEFAULT and RTLD_NEXT are defined by <dlfcn.h> only when _GNU_SOURCE was defined before including it.
```

Can we just replace these few lines with an unconditional `#define _GNU_SOURCE 1`?

**Review by dthain — CHANGES_REQUESTED** [review]

File: envtrace (hunk 1 above)
Comment (nkremerh) [reply envtrace]: /lib/libdl.so does not exist on my machine (/lib/libdl.so.2 does). I changed the LD_PRELOAD path to /lib64/libdl.so with seemingly no negative change in functionality.

**Comment by nkremerh** [comment]
This is ready for another look.

**Pull request merged**